### PR TITLE
Remove empty shapes

### LINKED ISSUE #15: Remove entities of area equal to 0 during serialization
For example:

- `Ball`s of radius = 0
- `Block`s with points in the same place

### PR DESCRIPTION
**Pull request by Fju**
Closes #15

I added a function to `src/utils/geom.ts` that calculates the area of a (convex) polygon given its vertices. This function is used to determine whether a Block is valid (area must be bigger than 0) in  [`src/models/Block.ts (line 115)`](https://github.com/getkey/ble/blob/e493414b11611ac41d488453e990b503bfb011bc/src/models/Block.ts#L115). If the Block is invalid it will be removed.

**TODO:**

- [x] Remove empty polygons
- [x] Add unit tests
- [x] ~Remove circles with radius 0~

### CODE REVIEW

**Comment by Fju** [comment]
Removing circles with radius smaller or equal to 0 is already implemented in [`src/models/Ball.ts (line 40)`](https://github.com/getkey/ble/blob/e493414b11611ac41d488453e990b503bfb011bc/src/models/Ball.ts#L40). Should this also be moved to a `cleanInvalid` function or should it remain the way it is?

**Comment by getkey** [comment]
It's not necessary to add `cleanInvalid` for circles because https://github.com/getkey/ble/blob/e493414b11611ac41d488453e990b503bfb011bc/src/models/Ball.ts#L17 ensures a `Ball` can't have a radius <= 0 already. It will throw an error if you try to.

`setRadius()` prevents this from happening. Maybe that's a bit weird, now that I think about it... I'd keep it though because it's better than an unhandled exception. 😛 

TL;DR: no need to do anything for circles 😄

**Comment by Fju** [comment]
Alright, I updated the TODO list :wink:

**Comment by getkey** [comment]
Thank you! Very nice that you added tests 😄

**Comment by getkey** [comment]
FYI I decided to add `polygonArea` to https://github.com/getkey/bombhopperio-level-tools/commit/f1a06a11f07080f7840c6150fdc56f3b2d67be30 so the backend can also use it to check the validity of the levels that are submitted. 🙂
